lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, David Kainz <dkainz@mgit.at> <dave.jokain@gmx.at>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: openssh_keypair
+author: "David Kainz (@lolcube) aka (@bsod)"
+version_added: "2.8"
+short_description: Generate OpenSSH private and public keys.
+description:
+    - "This module allows one to (re)generate OpenSSH private and public keys. It uses
+      ssh-keygen to generate keys. One can generate rsa, dsa, rsa1, ed25519
```

Comment:
Use `C(ssh-keygen)`, and also `... generate C(rsa), R(dsa), R(rsa1), C(ed25519), or C(ecdsa) private keys.` to improve formatting.

Reply:
And out of curiosity: what is `rsa1`? The ssh-keygen on my machine does not know about it (`unknown key type rsa1`); I've got OpenSSH_7.8p1 installed. From searching the internet, it looks like `rsa1` is a SSHv1 RSA key (and thus deprecated, as the whole SSHv1 protocol). If that's correct, you might want to mention that, and that not all implementations support it.

Reply:
`rsa1` is indeed the old, deprecated (but still supported on some platforms...) format for RSA keys.